### PR TITLE
scaffolder-backend,techdocs-node: remove global dependency mocks

### PR DESCRIPTION
**Pull request by Rugvip**
## Hey, I just made a Pull Request!

Went on a bit of a hunt to remove global module mocks, since they both don't work well along with execution from multiple projects at once (haven't quite figured out why yet, but suspecting they're not isolated enough), and also tend to make tests really hard to follow.

Starting with this proposal, but would suggest going as far as creating an ADR that we avoid using global dependency mocks.

No changesets needed as this just refactors tests.

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Review by jhaals — APPROVED** [review]

**Comment by benjdlambert** [comment]
@iamEAP could we get a +1 here?

**Review by iamEAP — APPROVED** [review]
Thanks!  Makes sense.

**Comment by iamEAP** [comment]
> _haven't quite figured out why yet, but suspecting they're not isolated enough_

^ Any insights on this appreciated!  I have a feeling our internal monorepo suffers from similar ills.
